CONTRIBUTING.md

```diff
@@ -114,6 +119,88 @@ There are several reasons we're using English while communicating in GitHub Issu
 
 ## Additional Notes
 
+### Commitlint
+
+In a repository with many contributors like this, it's important to understand
+what's going on in it and navigate between commits. Moreover, with various
+levels of contributors & their backgrounds, the commit messages can easily be
+confusing and the structure may follow different approaches.
+
+To overcome these issues, the maintainers adopt [commit conventions](https://www.conventionalcommits.org/en/)
+to allow contributors to add more semantic meaning to our git history. They use
+[commitlint](https://commitlint.js.org/) to lint the git commit messages.
+To enforce the conventions, the maintainers use a git hook to run the
+commitlint upon a git commit command. The git hook prevents the commit the
+staged files if the commit message failed to comply with the conventions.
+
+It's adviseable to read the [commit conventions](https://www.conventionalcommits.org/en/v1.0.0)
+briefly to further understand the other benefits from them. However, if you're
+having problems with it or what have you, you can use [this
+tool](https://commitlint.io/) to help.
+
+Furthermore, please refer to the list of [commit types](#commit-types) and
+[scopes](#commit-scopes) that we use to avoid adding a new scope that has similar
+meaning or is synonym to the existing one.
+
+## References
+
+### Conventional Commits
+
+The convention specification looks like this:
+
+```
+<type>[optional scope]: <description>
+
+[optional body]
+
+[optional footer(s)]
+```
+
+#### Commit types
+
+Here are what we use for the commit types:
+
+- `build`
+- `chore`
+- `docs`
+- `feat`
+- `fix`
+- `perf`
+- `refactor`
+- `style`
+- `test`
+
+#### Commit scopes
+
+Here are what we use for the scopes:
+
+- `ci`
+- `cms`
+- `cypress`
+- `deps`
+- `dialog`
+- `donasi`
+- `dx`
+- `e2e`
+- `education`
+- `eslint`
+- `faq`
+- `favicon`
+- `fetcher`
+- `fix`
+- `gtm`
+- `id`
+- `image`
+- `isoman`
+- `kodiak`
+- `layout`
+- `security`
+- `StackedLink`
+- `seo`
+- `test`
+- `text`
+- `ui`
```

Comment:
```suggestion
- `ci`
- `assets`
- `components`
  - `donasi`
  - `education`
  - `home`
  - `json-ld`
  - `kontak-darurat`
  - `layout`
  - `telemedicine`
  - `ui`
- `constants`
- `cms`
- `cypress`
- `data`
- `deps`
- `dx`
- `e2e`
- `fetcher`
- `pages`
  - `faq`
  - `isoman`
- `security`
- `seo`
- `ui`
```

@apraditya We can initially start with the top level directories. Bringing the scope levels to a higher level can help reduce the cognitive load on the contributors.

If there's a second level in a list item, only the lowest level gets to be used. But I don't know how to properly convey this.

Reply:
> @apraditya I tried narrowing down the scopes a bit. We can initially start with the top level directories. Bringing the scope levels to a higher level can help reduce the cognitive load on the contributors.

Agree

> If there's a second level in a list item, only the lowest level gets to be used. But I don't know how to properly convey this.

Agree with the first sentence. I don't have any preference at the moment. I use the suggestion and put the note about the second level.